src/deposit_monitor.rs

```diff
@@ -32,21 +34,31 @@ impl MonitoredDeposit {
 pub struct DepositMonitor {
     context: Context,
     monitored: HashMap<ScriptBuf, MonitoredDeposit>,
+    tx_hex_cache: LruCache<(Txid, BlockHash), String>,
 }
 
+// TODO: make cache size configurable
+// As for now numbers are chosen to keep cache size around 10MB
```

Comment:
I would reduce the cache size to 1MB, but I suppose this is okay too.

Reply:
We spoke offline about this, and settled on 4 MB in case we had a bitcoin block with a transaction that took up the entire block.

Reply:
[487f988](https://github.com/BitcoinL2-Labs/sPoX/pull/8/commits/487f988e619c9c36e560f6f8ffb7f62324ce1e2d)